tests/unit/test_token_response.py

```diff
@@ -0,0 +1,236 @@
+                             SDKTESTER1A_ID_ACCESS_TOKEN, get_client_data)
+from globus_sdk.auth.token_response import (
+    OAuthTokenResponse, OAuthDependentTokenResponse, _convert_token_info_dict)
+from globus_sdk.exc import GlobusOptionalDependencyError
+
+
+class OAuthTokenResponseTests(CapturedIOTestCase):
+
+    def setUp(self):
+        """
+        Creates a OAuthTokenResponse object with known data for testing
+        Sets up mock AuthClient for decoding id_tokens
+        """
+        super(OAuthTokenResponseTests, self).setUp()
+        # known token data for testing expected values
+        self.top_token = {  # valid id_token and access_token
+            "resource_server": "server1", "expires_in": 10, "scope": "scope1",
+            "refresh_token": "RT1", "other_tokens": [], "token_type": "1",
+            "id_token": SDKTESTER1A_NATIVE1_ID_TOKEN,
+            "access_token": SDKTESTER1A_ID_ACCESS_TOKEN}
+        self.other_token1 = {  # invalid id_token with valid access_token
+            "resource_server": "server2", "expires_in": 20, "scope": "scope2",
+            "refresh_token": "RT2", "other_tokens": [], "token_type": "2",
+            "id_token": "invalid_id_token",
+            "access_token": SDKTESTER1A_ID_ACCESS_TOKEN}
+        self.other_token2 = {  # valid id_token with invalid access_token
+            "resource_server": "server3", "expires_in": 30, "scope": "scope3",
+            "refresh_token": "RT3", "other_tokens": [], "token_type": "3",
+            "id_token": SDKTESTER1A_NATIVE1_ID_TOKEN,
+            "access_token": "invalid_access_token"}
+        self.top_token["other_tokens"] = [self.other_token1, self.other_token2]
+
+        # create the response
+        http_response = requests.Response()
+        http_response._content = six.b(json.dumps(self.top_token))
+        http_response.headers["Content-Type"] = "application/json"
+        self.response = OAuthTokenResponse(http_response)
+
+        # mock AuthClient
+        self.ac = mock.Mock()
+        self.ac.client_id = get_client_data()["native_app_client1"]["id"]
+        self.ac.get = mock.Mock(return_value={
+            "jwks_uri":
+            u"https://auth.globus.org/jwk.json"})
+
+    def test_convert_token_info_dict(self):
+        """
+        Converts known info dicts to confirm expected results
+        Confirms only refresh_token and token_type may be missing,
+        And expires_at is created correctly.
+        """
+        top_convert = _convert_token_info_dict(self.top_token)
+
+        # expected results for top token data
+        for key in ["scope", "access_token", "refresh_token", "token_type"]:
+            self.assertEqual(self.top_token[key], top_convert[key])
+
+        # sanity check expires at, assuming test will run well within 1 second
+        self.assertEqual(int(time.time()) + self.top_token["expires_in"],
+                         top_convert["expires_at_seconds"])
```

Comment:
As in the other case, let's account for the second-boundary case with `assertIn(..., (now - 1, now, now + 1))`

The assumption that we run within a second is okay, but this protects against rare, but confusing, spurious failures.
We could get by with `(now - 1, now)`, but it seems safer to include `now + 1` just to protect against some future refactor breaking this.